lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java

```diff
@@ -50,38 +51,43 @@ public static boolean hasServerRoot(Path filePath) {
         return hasServerRoot(filePath.toFile());
     }
 
-    public static boolean hasServerRoot(File xmlFile) {
+    private static boolean hasServerRoot(File xmlFile) {
         if (!xmlFile.exists() || xmlFile.length() == 0) {
             return false;
         }
         
         try {
             XMLInputFactory factory = getXmlInputFactory();
+            return hasSeverRootValues(factory,xmlFile);
+        } catch (Exception e) {
+            LOGGER.severe("Unable to access XML file "+ xmlFile.getAbsolutePath());
+        }
 
-            XMLEventReader reader = null;
+        return false;
+    }
 
-            try (FileInputStream fis = new FileInputStream(xmlFile)) {
-                reader = factory.createXMLEventReader(fis);
-                while (reader.hasNext()) {
-                    XMLEvent nextEvent = reader.nextEvent();
-                    if (nextEvent.isStartElement()) {
-                        return isServerElement(nextEvent);
-                    }
+    private static boolean hasSeverRootValues(XMLInputFactory factory, File xmlFile) {
+        XMLEventReader reader=null;
+        try {
+            FileInputStream fis = new FileInputStream(xmlFile);
+
+            reader = factory.createXMLEventReader(fis);
+            while (reader.hasNext()) {
+                XMLEvent nextEvent = reader.nextEvent();
+                if (nextEvent.isStartElement()) {
+                    return isServerElement(nextEvent);
                 }
-            } catch (XMLStreamException | FileNotFoundException e) {
-                LOGGER.severe("Error received trying to read XML file: " + xmlFile.getAbsolutePath()); 
-            } finally {
-                if (reader != null) {
-                    try {
-                        reader.close();
-                    } catch (Exception ignored) {   
-                    }
+            }
+        } catch (XMLStreamException | FileNotFoundException e) {
+            LOGGER.severe("Error received trying to read XML file: " + xmlFile.getAbsolutePath());
+        } finally {
+            if (reader != null) {
```

Comment:
If the `FileInputStream` is no longer in the try with resources, do we need to explicitly close it? Or does the `close()` on the `XmlReader` handle that?

Reply:
Adding close in finally block